libvast/src/type.cpp

```diff
@@ -2238,24 +2332,28 @@ merge(const record_type& lhs, const record_type& rhs,
                               "field {}; failed to merge {} and {}",
                               lfield.type.name(), rfield.type.name(),
                               rfield.name, lhs, rhs));
-              const auto lhs_attributes = lfield.type.attributes();
-              const auto rhs_attributes = rfield.type.attributes();
+              auto to_vector
+                = [](detail::generator<type::attribute_view>&& rng) {
+                    auto result = std::vector<type::attribute_view>{};
+                    for (auto&& elem : std::move(rng))
+                      result.push_back(std::move(elem));
+                    return result;
+                  };
```

Comment:
This is implemented again in the tests. I think it's fine to have a generic implementation in `generator.hpp`.

Reply:
I wouldn't wanna do so. In the future, we'll hopefully have `std::ranges::to<std::vector>`. It's only used in one place in all of libvast, and that is here. Usage in the tests should not imply making that function part of the API.

Reply:
I wouldn't want to `#include <ranges>` only for this.

Reply:
We only use it for `to_legacy_type` so we don't need it in the codebase long-term.